src/go/k8s/pkg/resources/statefulset_test.go

```diff
@@ -0,0 +1,139 @@
+package resources_test
+
+import (
+	"context"
+	"testing"
+
+	redpandav1alpha1 "github.com/vectorizedio/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	res "github.com/vectorizedio/redpanda/src/go/k8s/pkg/resources"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestEnsure(t *testing.T) {
```

Comment:
It's very slim and on point test! I only would like to use `Ensure` function to tests update scenarios. 

You could wrap current `TestEnsure` into it's own subtest. Whole table tests wrap in another subtest and call Ensure as this is our public interface.

Reply:
yeah I know, discussed this decision in another thread, I'll change it if we're against this...

Reply:
Per above discussion